gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java

```diff
@@ -277,27 +277,46 @@ public void close() throws IOException {
   /**
    * Creates the given topic with the given name.
    *
+   * Sample code:
+   * <pre><code>
+   * try (PublisherApi publisherApi = PublisherApi.createWithDefaults()) {
+   *   String formattedName = PublisherApi.formatTopicName("[PROJECT]", "[TOPIC]");
+   *   Topic response = publisherApi.createTopic(formattedName);
+   * }
+   * </code></pre>
+   *
    * <!-- manual edit -->
    * <!-- end manual edit -->
    *
-   * @param name The name of the topic. It must have the format
+   * @param formattedName The name of the topic. It must have the format
    * `"projects/{project}/topics/{topic}"`. `{topic}` must start with a letter,
    * and contain only letters (`[A-Za-z]`), numbers (`[0-9]`), dashes (`-`),
    * underscores (`_`), periods (`.`), tildes (`~`), plus (`+`) or percent
    * signs (`%`). It must be between 3 and 255 characters in length, and it
    * must not start with `"goog"`.
    * @throws com.google.api.gax.grpc.ApiException if the remote call fails
    */
-  public final Topic createTopic(String name) {
-    Topic request = Topic.newBuilder().setName(name).build();
-
+  public final Topic createTopic(String formattedName) {
+    TOPIC_PATH_TEMPLATE.validate(formattedName);
```

Comment:
First point: Yes, we would be capable of doing something like splitting into two params. I think we should be careful of that for a couple reasons though:
1. It takes us further away from the raw API surface. We want to track the API surface predictably.
2. In general use, people should be passing around the qualified resource names, and they should only have to be aware of the individual IDs when creating resources. I think it is better to be consistent and use the fully-qualified resource names everywhere.

Second point: I see your point about consistency in naming. Since we really don't have any control over Topic's fields, maybe we shouldn't try to use different naming approaches for our flattened parameters; that also kind of violates the principle I mentioned in the first point, which is that we should strive to adhere more closely to the raw API surface in this layer.